boot/duneboot.ml

```diff
@@ -1727,7 +1938,19 @@ let get_dependencies libraries =
     in
     let+ dependencies =
       let args = write_args "source_files" all_source_files in
-      ocamldep (mk_flags "-map" alias_files @ args)
+      ocamldep args
+    in
+    let lib_by_file =
+      List.concat_map libraries ~f:(fun (lib : Library.t) ->
+        List.rev_map lib.ocaml_files ~f:(fun f -> f, lib))
+      |> String.Map.of_list
+    in
+    let toplevel_modules =
+      List.map libraries ~f:(fun (lib : Library.t) -> lib.group) |> Group.merge_all
+    in
+    let path_by_obj =
+      List.fold_left libraries ~init:Module.Obj.Map.empty ~f:(fun acc (lib : Library.t) ->
+        Module.Obj.Map.union ~f:(fun _ _ _ -> assert false) acc lib.path_by_obj)
```

Comment:
Is this a todo for qualified?

Reply:
No, we expect all module object names to be unique.